packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -280,7 +281,12 @@ export default class NewVm extends BaseComponent {
   }
 
   componentDidMount() {
-    this._reset()
+    this._reset(() => {
+      const { template } = this.props
+      if (template !== undefined) {
+        this._initTemplate(this.props.template)
+      }
+    })
```

Comment:
Can you explain why you need to do this?

Reply:
Unfortunetly this PR https://github.com/vatesfr/xen-orchestra/pull/4500 was not enough in order to initialize template from URL query.
We used this [code](https://github.com/vatesfr/xen-orchestra/pull/4500/files#diff-a255ba66f114f6e11cd0862a0543b7dfR287) in order to initialize the template.
When the app is loaded, the template object corresponding to the query string ID is not loaded immediately, thereby initialize the template in `componentDidMount` was not a good idea, so we wait until we got something using `componentDidUpdate` and then we call `_initTemplate`.
But in the case when the template is loaded before `componentDidMount`, `componentDidUpdate` fails to watch if there was a change. That's why in this case, we need to initialize the template in `componentDidMount`.

Reply:
Indeed, thanks!